addons/web_editor/static/src/js/editor/image_processing.js

```diff
@@ -511,6 +511,31 @@ export function isImageSupportedForStyle(img) {
 
     return !isTFieldImg && !isEditableRootElement;
 }
+/**
+ * @param {HTMLImageElement} img
+ * @returns {Promise<Boolean>}
+ */
+export async function isImageCorsProtected(img) {
+    const src = img.getAttribute('src');
+    if (!src) {
+        return false;
+    }
+    let isCordProtected = false;
+    if (src && (!src.startsWith("/") || /\/web\/image\/\d+-redirect\//.test(src))) {
+        // The `fetch()` used later in the code might fail if the image is
+        // CORS protected. We check upfront if it's the case.
+        // Two possible cases:
+        // 1. the `src` is an absolute URL from another domain.
+        //    For instance, abc.odoo.com vs abc.com which are actually the
+        //    same database behind.
+        // 2. A "attachment-url" which is just a redirect to the real image
+        //    which could be hosted on another website.
+        isCordProtected = await fetch(src)
```

Comment:
Maybe you could try not fetching the whole image to have the info? Maybe via `fetch(url, {method: 'HEAD'})`?

Reply:
Very good catch :+1: